File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/3372-longest-strictly-increasing-or-strictly-decreasing-subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        mxInc = 1
        mxDec = 1
        ans = 0
        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                mxInc += 1
                mxDec = 1
            elif nums[i] < nums[i-1]:
                mxInc = 1
                mxDec += 1
            else:
                mxInc = mxDec = 1
            ans = max(mxInc, mxDec, ans)
        return ans if len(nums) > 1 else 1

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/3372-longest-strictly-increasing-or-strictly-decreasing-subarray.java
class Solution {
    public int longestMonotonicSubarray(int[] nums) {
        int Inc = 1;
        int Dec = 1;
        int ans = 0;
        int n = nums.length;
        if(n==1){
            return 1;
        }
        for(int i= 1; i< n; i++){
            if(nums[i] < nums[i-1]){
                Dec += 1;
                Inc =1;   
            }else if(nums[i] > nums[i-1]){
                Dec =1;
                Inc += 1;
            }else{
                Dec = 1;
                Inc = 1;
            }
            ans = Math.max(ans, Math.max(Inc, Dec));
        }
        return ans;
    }
}